exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/DrillCompoundIdentifier.java

```diff
@@ -69,31 +70,38 @@ public void addIndex(int index, SqlParserPos pos){
     }
   }
 
-  public SqlNode getAsSqlNode(){
-    if(ids.size() == 1){
+  public SqlNode getAsSqlNode(Set<DrillCompoundIdentifier> fullSchemasSet) {
```

Comment:
Calcite has a concept of a "namespace" that abstracts what columns are available in a table or sub-query. I think you should be using that rather than looking at the structure of the parse tree.

There's a lot of code here, and it seems to duplicate (in a less general way) what is being done in Calcite. It's technical debt, and let me explain how it will bite Drill. I am working right now on https://issues.apache.org/jira/browse/CALCITE-1208 and making some significant changes to how name-resolution works. When I check in CALCITE-1208 there's a strong chance that the code in this PR will break, and as a result, it will take Drill even longer to get back onto Calcite master branch.